paper-badge.html

```diff
@@ -116,6 +132,25 @@
         label: {
           type: String,
           observer: '_labelChanged'
+        },
+
+        /**
+         * An iron-icon ID. When given, the badge content will use an
+         * `<iron-icon>` element displaying the given icon ID rather than the
+         * label text. However, the label text will still be used for
+         * accessibility purposes.
+         */
+        icon: {
+          type: String,
+          value: ''
```

Comment:
I tried this before resorting to the computed property, and the result was that the `iron-icon` element would not have the `hidden` attribute. I spent the better part of the morning investigating why this is, and I think the offending code is here: https://github.com/Polymer/polymer/blob/cfa6d51479c75e744ec8aa5a5d78968c5897eb7d/src/lib/annotations/annotations.html#L326-L328

```javascript
var literal = this._literalFromParts(parts);
if (literal && kind == 'attribute') {
  node.setAttribute(name, literal);
}
```

Basically if there is no literal value associated with the attribute, it skips setting the attribute altogether, without taking whether or not `negate` is set into account. It _does_ work with computed properties, however, because the attribute value is corrected when the computed effects are applied and the computed value is serialized to the attribute.

Another curious case is why the badges that _do_ contain an icon display correctly. It turns out that for the element where `hidden$="{{icon}}"` it actually sets the hidden prop to the literal value of that icon. However, when I use `{{!!icon}}` it does _not_ work because the annotation parser looks at the first `!` and thinks it's a negation.

For more context, there's also [this stackoverflow question](), the answer which recommends using a dom-if, and the question talks about using a computed prop as a workaround. I think that's the only way to accomplish this for now. This could be fixed using a number of strategies, for example:
* Looking at `negate` within `parseNodeAttributeAnnotations` and if it's set and there's no literal, assign the attribute name as the value (e.g. `hidden="hidden"`) or something like that.
* Use a dummy "sentinel" element, and if there's no literal and `negate` is set, assign `!dummyElement[attrName]` as the value, e.g.

```javascript
var dummy = document.createElement(node.tagName);
if (kind == 'attribute' && (literal || !literal && negate)) {
  node.setAttribute(name, literal ? literal : !dummy[name]);
}
// ...
```
* Keep a list of attributes and their default values and reference that within this method.

Happy to open an issue on Polymer/polymer to discuss this further if there isn't one open already.